src/Mvc/Mvc.Razor/src/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -322,9 +320,23 @@ private static string CreateTrimmedString(string input, int start)
         return input.Substring(start, len);
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private static bool IsCharWhitespace(char ch)
     {
-        return ValidAttributeWhitespaceChars.AsSpan().IndexOf(ch) != -1;
+        if (ch > 32)
+        {
+            return false;
+        }
+
+        // Valid whitespace characters defined by the HTML5 spec.
+        const long BitMask =
+            (1L << '\t')    //  9
+          | (1L << '\n')    // 10
+          | (1L << '\f')    // 12
+          | (1L << '\r')    // 13
+          | (1L << ' ');    // 32
+
+        return (BitMask & (1L << ch)) != 0;
```

Comment:
The operation is done as `long`, so codegen for 32-bit platforms won't be as ideal as for 64-bit targets. Do we care? If so, I can change the code to also honor 32-bit.